python/tank_vendor/shotgun_authentication/ui/login_dialog.py

```diff
@@ -0,0 +1,173 @@
+# Copyright (c) 2013 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+"""
+Login dialog for authenticating to a Shotgun server.
+"""
+
+from . import resources_rc
+from . import ui_login_dialog
+from .. import session_cache
+from ..errors import AuthenticationError
+from .qt_abstraction import QtGui, QtCore
+
+
+class LoginDialog(QtGui.QDialog):
+    """
+    Dialog for getting user crendentials.
+    """
+    def __init__(self, title, is_session_renewal, hostname="", login="", fixed_host=False, http_proxy=None, pixmap=None, parent=None):
+        """
+        Constructs a dialog.
+
+        :param title: Title of this dialog.
+        :param is_session_renewal: Boolean indicating if we are renewing a session or authenticating a user from scratch.
+        :param hostname: The string to populate the site field with. Defaults to "".
+        :param login: The string to populate the login field with. Defaults to "".
+        :param fixed_host: Indicates if the hostname can be changed. Defaults to False.
+        :param http_proxy: The proxy server to use when testing authentication. Defaults to None.
+        :param pixmap: QPixmap to show in the dialog (defaults to the Shotgun logo)
```

Comment:
We can get rid of this for the first version of this - if we need it later, we can always bring it back!